ambari-common/src/main/python/ambari_commons/repo_manager/command_repository.py

```diff
@@ -0,0 +1,262 @@
+"""
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ambari_commons.os_check import OSCheck
+import ambari_simplejson as json
+
+__all__ = ["RepositoryUtil", "CommandRepository", "UBUNTU_REPO_COMPONENTS_POSTFIX"]
+
+# components_list = repoName + postfix
+UBUNTU_REPO_COMPONENTS_POSTFIX = "main"
+
+
+class RepositoryUtil:
+  def __init__(self, config, tags_to_skip=()):
+    """
+    :type config dict
+    :type tags_to_skip set
+    """
+    from resource_management.libraries.functions.is_empty import is_empty
+
+    self.tags_to_skip = tags_to_skip
+
+    # repo templates
+    repo_file = config['repositoryFile']
+    repo_rhel_suse = config['configurations']['cluster-env']['repo_suse_rhel_template']
+    repo_ubuntu = config['configurations']['cluster-env']['repo_ubuntu_template']
+
+    if is_empty(repo_file):
+      return
+
+    self.template = repo_rhel_suse if OSCheck.is_redhat_family() or OSCheck.is_suse_family() else repo_ubuntu
+    self.command_repository = CommandRepository(repo_file)
+
+  def create_repo_files(self):
+    """
+    Creates repositories in a consistent manner for all types
+    :return: a dictionary with repo ID => repo file name mapping
+    """
+    from resource_management.core.logger import Logger
+    from resource_management.libraries.resources.repository import Repository
+
+    if self.command_repository.version_id is None:
+      raise ValueError("The command repository was not parsed correctly")
+
+    if 0 == len(self.command_repository.items):
+      Logger.warning(
+        "Repository for {0}/{1} has no repositories.  Ambari may not be managing this version.".format(
+          self.command_repository.stack_name, self.command_repository.version_string))
+      return {}
+
+    repo_files = {}
+    for repository in self.command_repository.items:
+      if repository.repo_id is None:
+        raise ValueError("Repository with url {0} has no id".format(repository.base_url))
+
+      if self.tags_to_skip & repository.tags:
```

Comment:
Does & work on set + tuple or a set + list?

Reply:
nope, there passes set + set and there is a check for this

Reply:
p.s. this is code existed before my patch and is just moved to new location

Reply:
But the default value of tags_to_skip is an empty tuple.

Reply:
no idea, check the previous code https://github.com/apache/ambari/blame/trunk/ambari-common/src/main/python/resource_management/libraries/functions/repository_util.py#L67

for what was the exact idea behind this, could be asked @aonishuk

Reply:
ok, then we can just force `tags_to_skip` to be a set, which would check if incoming argument value is compatible with set as well.

The default value will stay as empty tuple, as it is a bad practice to make a default value matable